barretenberg/cpp/src/barretenberg/crypto/merkle_tree/indexed_tree/indexed_leaf.hpp

```diff
@@ -107,7 +107,7 @@ struct PublicDataLeafValue {
 
     fr get_key() const { return slot; }
 
-    bool is_empty() const { return slot == fr::zero(); }
+    bool is_empty() const { return slot == fr::zero() && value == fr::zero(); }
```

Comment:
This was triggering a bug: slot 0 was being updated to the value `128` in the avm_tree.test.ts and the insertion code thought that the write was empty, since it was only checking the slot.